### PR TITLE
Update plugin for Pipeline support

### PR DESCRIPTION
**Pull request by ConstantineXVI**
Ported to newer analysis-core and adapted to be pipeline-friendly

### CODE REVIEW

**Comment by ConstantineXVI** [comment]
Have not yet verified this won't blow away existing results. Working up a 'live' environment to ensure this isn't the case at some point today or tomorrow.

**Comment by bitwiseman** [comment]
It is great that you're doing this.  Did you also add a definition for a step for this publisher like it does  
[https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/300fe6c02b41f072e50a501cfec3e2f425048446/src/main/java/org/jenkinsci/plugins/workflow/steps/ArtifactArchiverStep.java#L37-L53](here)?

**Comment by ConstantineXVI** [comment]
@bitwiseman Was planning on it. Haven't noticed a particular naming/param convention for other reporting plugins (particularly HealthAwarePublisher derived ones), planned to do a bit more research on that front to be sure I'm not violating any expectations.

File: src/main/java/hudson/plugins/brakeman/BrakemanPublisher.java [hunk 1]
```diff
@@ -60,7 +47,7 @@ public BrakemanPublisher(final String healthy, final String unHealthy, final Str
 				final String failedTotalAll, final String failedTotalHigh, final String failedTotalNormal, final String failedTotalLow,
 				final String failedNewAll, final String failedNewHigh, final String failedNewNormal, final String failedNewLow,
 				final boolean canRunOnFailed, final boolean shouldDetectModules, final boolean canComputeNew, final String outputFile) {
-
+			// TODO use non-deprecated constructor
```
Comment (bitwiseman) [inline src/main/java/hudson/plugins/brakeman/BrakemanPublisher.java]: Consider filing an bug for this on the jenkins JIRA and add the link to it here.  TODO's are easy to overlook.

**Comment by bitwiseman** [comment]
@ConstantineXVI There doesn't appear to be one at this time, so naming is a bit all over the place.  I could make some suggestions but they'd be only my personal tastes. 😄

**Comment by ConstantineXVI** [comment]
Hm, I'm not sure how to add in a pipeline step that reuses the "classic"[?] publisher class instead of reimplementing it. Entirely possible I'm just not finding the right docs on this.

File: src/main/java/hudson/plugins/brakeman/BrakemanPublisher.java (hunk 1 above)
Comment (ConstantineXVI) [inline src/main/java/hudson/plugins/brakeman/BrakemanPublisher.java]: Normally better at leaving my TODOs out of commits. Resolved.

**Comment by bitwiseman** [comment]
Would this work? 
https://github.com/jenkinsci/htmlpublisher-plugin/blob/73fdcc6e0a668d27bc390567f95cc8c9ec445cc1/src/main/java/htmlpublisher/workflow/PublishHTMLStep.java

**Comment by ConstantineXVI** [comment]
Seems it would, yes

**Comment by bitwiseman** [comment]
Very cool.

**Comment by ConstantineXVI** [comment]
Confirmed old results and builds still work. I'm willing to call this sane.

**Comment by presidentbeef** [comment]
:heart_eyes:

**Comment by ConstantineXVI** [comment]
Does anything else need to be done to get this merged?

**Comment by presidentbeef** [comment]
Probably not, I just need to test it out myself.

**Comment by ConstantineXVI** [comment]
*ping

**Comment by presidentbeef** [comment]
Sorry for the delay, I was waiting for maven to finish :racehorse:

I noticed there's no Brakeman warning trend graph for the pipeline project...is this expected?

Otherwise, looks fine to me.

**Comment by ConstantineXVI** [comment]
I'll give that a peek. Suspect I could be setting better defaults for the DSL step.

**Comment by ConstantineXVI** [comment]
OK, from my testing this looks like it might be an upstream issue/unimplemented feature in analysis-core with pipelines. May fix it anyway, but don't think I can do it here.

**Comment by presidentbeef** [comment]
Thank you so much for these updates!

Are you interested in being a maintainer for this plugin?

**Comment by ConstantineXVI** [comment]
Sure.

**Comment by presidentbeef** [comment]
Okay, I'll just do this next release and then get you set up.

**Comment by ConstantineXVI** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-31202 (analysis-core pipeline trend graphs) already in progress by Ulli, just to close that loop

**Pull request merged**